src/ngraph_builder.cc

```diff
@@ -1351,6 +1351,126 @@ static Status TranslateConv2DBackpropInputOp(
   return Status::OK();
 }
 
+// Translate DepthToSpace op
+static Status TranslateDepthToSpaceOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input));
+
+  // Get the attributes
+  int64 block_size;
+  std::string tf_data_format;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "block_size", &block_size));
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
+
+  ng::Shape input_shape = ng_input->get_shape();
+  std::map<std::string, int> format_to_int_map = {
```

Comment:
We can use if-else loop and avoid this map. Not really sure of the advantage of using this map

Reply:
I see you are using it again. :)

Reply:
Hi @shresthamalik, I chose map without any particular reason. But I feel like it is more readable than `if-else`. In terms of performance, the switch statement could be the same as `if-else`. Are you concerned about this one more data structure? I can change to `if-else` as well.